### PR TITLE
✨(dashboard) add authentication system and custom user model

### PR DESCRIPTION
**Pull request by ssorin**
Introduced a new user model `DashboardUser` and related configurations for user management, authentication, and admin interface. 
Implemented user login functionality and protected menu options based on user authentication status.

### CODE REVIEW

File: src/dashboard/templates/registration/login.html [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                id="credentials"
+                                aria-labelledby="credentials-messages">
+                        <div class="fr-fieldset__element">
+                          <span class="fr-hint-text">Sauf mention contraire, tous les champs sont obligatoires.</span>
+                        </div>
+                        <div class="fr-fieldset__element">
+                          <div class="fr-input-group">
+                            <label class="fr-label" for="id_username">
+                              Identifiant
+                              <span class="fr-hint-text">Format attendu : nom@domaine.fr</span>
```
Comment (jmaupetit) [inline src/dashboard/templates/registration/login.html:49]:
```suggestion
                              <span class="fr-hint-text">Format attendu : myusername</span>
```

File: src/dashboard/templates/registration/login.html
Comment (jmaupetit) [inline src/dashboard/templates/registration/login.html:1]: Don't forget to make content translatable.

**Review by jmaupetit — COMMENTED** [review]
Looks ok to me :+1:

File: src/dashboard/apps/users/__init__.py
Comment (jmaupetit) [inline src/dashboard/apps/users/__init__.py:1]: As discussed, may be move the `users` app to `auth`.

File: src/dashboard/apps/auth/apps.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from django.apps import AppConfig
+from django.utils.translation import gettext_lazy as _
+
+
+class UsersConfig(AppConfig):
+    """Auth app config."""
+
+    default_auto_field = "django.db.models.BigAutoField"
+    name = "apps.auth"
+    label = "apps_auth"
```
Comment (jmaupetit) [inline src/dashboard/apps/auth/apps.py:12]:
```suggestion
    label = "dashboard_apps_auth"
```

File: src/dashboard/apps/auth/apps.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+"""Dashboard auth app base config."""
+
+from django.apps import AppConfig
+from django.utils.translation import gettext_lazy as _
+
+
+class UsersConfig(AppConfig):
```
Comment (jmaupetit) [inline src/dashboard/apps/auth/apps.py:7]:
```suggestion
class AuthConfig(AppConfig):
```

File: src/dashboard/apps/home/apps.py [hunk 4]
```diff
@@ -4,7 +4,8 @@
 
 
 class HomeConfig(AppConfig):
-    """Home app config."""
+    """home app config."""
```
Comment (jmaupetit) [inline src/dashboard/apps/home/apps.py:7]: Title case is better for a docstring right?

File: src/dashboard/readme.md [hunk 5]
```diff
@@ -36,6 +36,21 @@ Display dashboard logs
...
 
+## Project specific naming convention
+
+For each Django application, the application config label 
+(present in my_apps.apps.MyAppConfig) must be of type `qcd_myapp`.
+
+i.e. for the `home` app:  
+```python
+class HomeConfig(AppConfig):
+    """home app config."""
```
Comment (jmaupetit) [inline src/dashboard/readme.md:47]: :grimacing: 

```suggestion
    """Home app config."""
```

**Review by jmaupetit — COMMENTED** [review]
Almost done!

File: src/dashboard/templates/registration/login.html [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                        <div class="fr-fieldset__element">
+                          <span class="fr-hint-text">
+                            {% trans "Unless otherwise stated, all fields are required." %}                            
+                          </span>
+                        </div>
+                        <div class="fr-fieldset__element">
+                          <div class="fr-input-group">
+                            <label class="fr-label" for="id_username">
+                              {% trans "Identifier" %}
+                              <span class="fr-hint-text">{% trans "Expected format: name@domain.fr" %} </span>
```
[the remark this answers is not in the document]
Comment (ssorin) [reply src/dashboard/templates/registration/login.html]: ✅

File: src/dashboard/apps/auth/apps.py (hunk 3 above)
Comment (ssorin) [reply src/dashboard/apps/auth/apps.py]: ✅

File: src/dashboard/apps/home/apps.py (hunk 4 above)
Comment (ssorin) [reply src/dashboard/apps/home/apps.py]: ✅

File: src/dashboard/readme.md (hunk 5 above)
Comment (ssorin) [reply src/dashboard/readme.md]: ✅

File: src/dashboard/templates/registration/login.html (hunk 1 above)
Comment (ssorin) [reply src/dashboard/templates/registration/login.html]: ✅

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**